pkg/sql/importer/read_import_base.go

```diff
@@ -112,28 +112,22 @@ func runImport(
 	var summary *kvpb.BulkOpSummary
 	group.GoCtx(func(ctx context.Context) error {
 		summary, err = ingestKvs(ctx, flowCtx, spec, progCh, kvCh)
-		if err != nil {
-			return err
-		}
-		var prog execinfrapb.RemoteProducerMetadata_BulkProcessorProgress
-		prog.ResumePos = make(map[int32]int64)
-		prog.CompletedFraction = make(map[int32]float32)
-		for i := range spec.Uri {
-			prog.CompletedFraction[i] = 1.0
-			prog.ResumePos[i] = math.MaxInt64
-		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case progCh <- prog:
-			return nil
-		}
+		return err
 	})
 
 	if err = group.Wait(); err != nil {
 		return nil, err
 	}
 
+	var prog execinfrapb.RemoteProducerMetadata_BulkProcessorProgress
+	prog.ResumePos = make(map[int32]int64)
+	prog.CompletedFraction = make(map[int32]float32)
+	for i := range spec.Uri {
+		prog.CompletedFraction[i] = 1.0
+		prog.ResumePos[i] = math.MaxInt64
+	}
+	progCh <- prog
```

Comment:
I think you want this in the select as it was before

Reply:
That isn't a concern I know anything about so you'll have to educate me.   Are you saying the progCh consumer goroutine could stop reading when a context is canceled so our send would hang or panic?  Currently it doesn't look like that's possible (the goroutine reading from that channel doesn't check for cancelation) but I guess you mean that select'ing over the context in addition to writing the channel is good hygiene so we should do it anyways?

Reply:
yeah, I generally treat any send or recv on a ch that doesn't also select on ctx done as suspect until proven otherwise, but if you're verified this one is known to be okay since the consumer cannot exit before the sender under any circumstance then carry on.

My personal pref is to just always do the select since sometimes someone adds a `return err` or something to the other side than then it can exit or something, but as long as you're convinced it is safe as-is, I'm happy.